compiler/util/parser_types.py

```diff
@@ -12,72 +12,137 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Various types shared through multiple passes of parsing.
+"""Types related to the LR(1) parser.
 
-This module contains types used as interfaces between parts of the Emboss front
-end.  These types do not really "belong" to either the producers or consumers,
-and in a few cases placing them in one or the other creates unnecessary
-dependencies, so they are defined here.
+This module contains types used by the LR(1) parser, which are also used in
+other parts of the compiler: 
+
+    SourcePosition: a position (zero-width) within a source file.
+    SourceLocation: a span within a source file.
+    Production: a grammar production.
+    Token: a token; lr1.Parser operates on lists of tokens.
 """
 
 import collections
-from compiler.util import ir_data
-
-
-def _make_position(line, column):
-    """Makes an ir_data.Position from line, column ints."""
-    if not isinstance(line, int):
-        raise ValueError("Bad line {!r}".format(line))
-    elif not isinstance(column, int):
-        raise ValueError("Bad column {!r}".format(column))
-    return ir_data.Position(line=line, column=column)
 
+PositionTuple = collections.namedtuple(
+    "PositionTuple", ["line", "column"], defaults=[0, 0]
+)
 
-def _parse_position(text):
-    """Parses an ir_data.Position from "line:column" (e.g., "1:2")."""
-    line, column = text.split(":")
-    return _make_position(int(line), int(column))
 
+class SourcePosition(PositionTuple):
+    """A zero-width position within a source file."""
```

Comment:
Perhaps document that `(0,0)` is the default position, and is considered invalid/falsey/unset.

Reply:
Done.